fuel-p2p/src/service.rs

```diff
@@ -332,4 +375,67 @@ mod tests {
             };
         }
     }
+
+    #[tokio::test]
+    async fn request_response_works() {
```

Comment:
should we have a negative test that verifies the outbound channel mappings are cleaned up when a timeout occurs?

Reply:
added one